charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-node-windows-hostprocess.yaml

```diff
@@ -0,0 +1,161 @@
+{{- if and (.Values.windows.enabled) (eq .Values.windows.useHostProcessContainers true) }}
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: {{ .Values.windows.dsName }}
+  namespace: {{ .Release.Namespace }}
+{{ include "azuredisk.labels" . | indent 2 }}
+spec:
+  updateStrategy:
+    rollingUpdate:
+      maxUnavailable: {{ .Values.node.maxUnavailable }}
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app: {{ .Values.windows.dsName }}
+  template:
+    metadata:
+{{ include "azuredisk.labels" . | indent 6 }}
+        app: {{ .Values.windows.dsName }}
+{{- with .Values.windows.podLabels }}
+{{ toYaml . | indent 8 }}
+{{- end }}
+    spec:
+      serviceAccountName: {{ .Values.serviceAccount.node }}
+{{- with .Values.windows.tolerations }}
+      tolerations:
+{{ toYaml . | indent 8 }}
+{{- end }}
+      nodeSelector:
+        kubernetes.io/os: windows
+      affinity:
+        nodeAffinity:
+{{ toYaml .Values.windows.nodeAffinity | indent 10 }}
+      priorityClassName: system-node-critical
+      {{- include "azuredisk.pullSecrets" . | indent 6 }}
+      securityContext:
+        windowsOptions:
+          hostProcess: true
+          runAsUserName: "NT AUTHORITY\\SYSTEM"
+      hostNetwork: true
+      initContainers:
```

Comment:
is it possible to combine these settings with original non-hostprocess deployment? looks like the main change is `initContainers`, otherwise we need to maintain two copies of windows daemonset in long term.

Reply:
Are you suggesting to update the csi-azuredisk-node-windows.yaml deployment use host-process and not introduce another file?

I don't think AKS supports host-process containers right now but if we are OK with saying only older existing versions
of the driver work on AKS I can do this.

I was thinking we could have both types of deployments exist at the same time for a while in case we find any issues with the host-process container deployments and then delete them at a later time.
Let me know what you think.

Reply:
correct, update `charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-node-windows.yaml` since chart config supports if-else, that would be easier to maintain in the future. And for `deploy/csi-azuredisk-node-windows.yaml`, looks like we have to create a new file for host-process containers

Reply:
Sounds good. I will do this today.

Reply:
I merged the two windows node helm template files - PTAL.
I left them in a sperate commit in case reviewer do not like the changes (I think it makes the helm templates a little bit more complicated).

Reply:
Hmm, it looks like there is a check to ensure the number of .yaml files in the chart template directory matches the number of files in /deploy

```
yaml file num(10) under deploy/ not equal to num(9) under charts/latest/azuredisk-csi-driver/templates
make: *** [Makefile:85: verify] Error 1
```